src/type-hierarchy.ts

```diff
@@ -82,6 +83,43 @@ class TypeHierarchyTreeItem extends vscode.TreeItem {
   }
 }
 
+class TypeHierarchyFeature implements vscodelc.StaticFeature {
+  private serverSupportsTypeHierarchy = false;
+  private state: vscodelc.State;
+
+  constructor(client: vscodelc.LanguageClient,
+              context: vscode.ExtensionContext) {
+    new TypeHierarchyProvider(context, client);
+    client.onDidChangeState(stateChange => {
+      this.state = stateChange.newState;
+      this.recomputeEnableTypeHierarchy();
+    });
+  }
+
+  fillClientCapabilities(capabilities: vscodelc.ClientCapabilities) {}
+
+  initialize(capabilities: vscodelc.ServerCapabilities,
+             documentSelector: vscodelc.DocumentSelector|undefined) {
+    const serverCapabilities: vscodelc.ServerCapabilities&
+        {typeHierarchyProvider?: boolean} = capabilities;
+    if (serverCapabilities.typeHierarchyProvider) {
+      this.serverSupportsTypeHierarchy = true;
+      this.recomputeEnableTypeHierarchy();
+    }
+  }
+
+  private recomputeEnableTypeHierarchy() {
+    if (this.state == vscodelc.State.Running) {
+      vscode.commands.executeCommand(
+          'setContext', 'extension.vscode-clangd.enableTypeHierarchy',
```

Comment:
Looks like we are missing the `Starting` state enum, I think it is fine.

Do you know what's the default value of `extension.vscode-clangd.enableTypeHierarchy` if there is no ` vscode.commands.executeCommand` being executed yet? probably undefined?

Reply:
Yes, I expect it's undefined (and therefore falsy when evaluated as a boolean).